libp2p/connmanager.nim

```diff
@@ -67,16 +66,11 @@ type
   PeerEventHandler* =
     proc(peerId: PeerId, event: PeerEvent): Future[void] {.gcsafe, raises: [Defect].}
 
-  MuxerHolder = object
-    muxer: Muxer
-    handle: Future[void]
-
   ConnManager* = ref object of RootObj
     maxConnsPerPeer: int
     inSema*: AsyncSemaphore
     outSema*: AsyncSemaphore
-    conns: Table[PeerId, HashSet[Connection]]
-    muxed: Table[Connection, MuxerHolder]
+    muxed: Table[PeerId, seq[Muxer]]
```

Comment:
how about non-muxed connections, like quic?

Reply:
Quic is still muxed, it's just native, so the muxer will just call the native procedures
https://github.com/status-im/nim-libp2p/pull/725/files#diff-e6732ffe4393737472ba25d1a6f85933cd5b5b2c85c7356a9684e05e4de8eabcR87-R92

I think it's a safe assumption to make that every libp2p is going to be muxed (otherwise we loose a big benefit of libp2p)

Reply:
I guess it can be bypassed... should be fine, I think overall it makes things a lot simpler.